test/lib/mayaUsd/utils/test_ShaderGenUtils.cpp

```diff
@@ -151,3 +153,54 @@ TEST(ShaderGenUtils, topoGraphAPI_defaultgeomprop)
     }
 }
 #endif
+
+TEST(ShaderGenUtils, lobePruner)
+{
+    namespace sgu = MaterialXMaya::ShaderGenUtil;
+
+    auto testPath = mx::FilePath(MATERIALX_TEST_DATA);
+
+    auto library = mx::createDocument();
+    ASSERT_TRUE(library != nullptr);
+    auto searchPath = PXR_NS::HdMtlxSearchPaths();
+    mx::loadLibraries({}, searchPath, library);
+
+    auto doc = mx::createDocument();
+    doc->importLibrary(library);
+
+    auto lobePruner = sgu::LobePruner::create();
+    lobePruner->setLibrary(doc);
+
+    const auto attrVec = lobePruner->getOptimizedAttributeNames(
+        doc->getNodeDef("ND_standard_surface_surfaceshader"));
+    ASSERT_FALSE(attrVec.empty());
+    ASSERT_TRUE(std::is_sorted(attrVec.begin(), attrVec.end()));
+    ASSERT_TRUE(std::lower_bound(attrVec.begin(), attrVec.end(), "subsurface") != attrVec.end());
+
+    const auto node = doc->addNode("standard_surface", "bob", "surfaceshader");
+
+    std::string optimizedCategory;
+    ASSERT_TRUE(lobePruner->getOptimizedNodeCategory(*node, optimizedCategory));
+    // An x means can not optimize on that attribute
+    // A 0 means we optimized due to this value being zero
```

Comment:
[Nitpick] Could be helpful to have a similar comment in `LobePruner.cpp` to explain the naming convention. Maybe there already is one and I missed it and it's not too difficult to figure it out just by reading the code, but it could save some time.